runner/internal/shim/api/server.go

```diff
@@ -29,12 +30,13 @@ type ShimServer struct {
 	version string
 }
 
-func NewShimServer(address string, runner TaskRunner, version string) *ShimServer {
+func NewShimServer(ctx context.Context, address string, runner TaskRunner, version string) *ShimServer {
 	r := api.NewRouter()
 	s := &ShimServer{
 		HttpServer: &http.Server{
-			Addr:    address,
-			Handler: r,
+			Addr:        address,
+			Handler:     r,
+			BaseContext: func(l net.Listener) context.Context { return ctx },
```

Comment:
Does it mean the same context will be used for all requests? I assume it will be a problem if we need to store per-request values in context.

Reply:
If I understand correctly, base context (as its name implies) is not used directly but to derive a new per-request context: https://github.com/golang/go/blob/194de8fbfaf4c3ed54e1a3c1b14fc67a830b8d95/src/net/http/server.go#L3318-L3328